From: Hongfu Li <lihongfu@kylinos.cn>
Subject: mm/swap: fix swap_cluster_lock() !CONFIG_SWAP stub signature mismatch
Date: Fri, 17 Jul 2026 15:11:04 +0800

The !CONFIG_SWAP stub for swap_cluster_lock() has mismatched prototype: it
has an extra unused irq argument and uses pgoff_t instead of unsigned long
for offset.  All callers are under CONFIG_SWAP so the extra parameter is
dead.

Delete the unused stub function entirely.

Link: https://lore.kernel.org/20260717071104.73467-1-hongfu.li@linux.dev
Signed-off-by: Hongfu Li <lihongfu@kylinos.cn>
Reviewed-by: Baoquan He <baoquan.he@linux.dev>
Acked-by: Kairui Song <kasong@tencent.com>
Cc: Barry Song <baohua@kernel.org>
Cc: Chris Li <chrisl@kernel.org>
Cc: Hongfu Li <lihongfu@kylinos.cn>
Cc: Kemeng Shi <shikemeng@huaweicloud.com>
Cc: Nhat Pham <nphamcs@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/swap.h |    5 -----
 1 file changed, 5 deletions(-)

--- a/mm/swap.h~mm-swap-fix-swap_cluster_lock-config_swap-stub-signature-mismatch
+++ a/mm/swap.h
@@ -358,11 +358,6 @@ void swap_update_readahead(struct folio
 			   unsigned long addr);
 
 #else /* CONFIG_SWAP */
-static inline struct swap_cluster_info *swap_cluster_lock(
-	struct swap_info_struct *si, pgoff_t offset, bool irq)
-{
-	return NULL;
-}
 
 static inline struct swap_cluster_info *swap_cluster_get_and_lock(
 		struct folio *folio)
_
